### PR TITLE
fix(logging): configuration debug call

### PR DESCRIPTION
**Pull request by therve**
Trying to run with DD_TRACE_DEBUG set and the agent writer fails with `TypeError: Object of type AgentWriter is not JSON serializable`. This fixes it.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by gnufede — DISMISSED** [review]

**Review by mabdinur — COMMENTED** [review]
Nice catch. We should also add a fix release note: https://github.com/DataDog/dd-trace-py/blob/1.x/docs/releasenotes.rst#release-note-sections

**Comment by mabdinur** [comment]
Also what versions of ddtrace are impacted by this bug? We might need to backport this fix to the older release lines and start a patch release: 1.16, 1.15, 1.14, .....

**Comment by therve** [comment]
Did you close that one on purpose?

**Comment by majorgreys** [comment]
Now that 1.16 is released, 1.14 is not maintained. 

@therve Do you need a patch release on 1.14 with this fix or is upgrading to 1.15 or 1.16 an option?

**Comment by therve** [comment]
1.16 is fine! I was just surprised to not see the fix landed on any branch.

**Comment by majorgreys** [comment]
Oops, I'm sorry, I thought this was a backport PR. Fixing now!

**Review by majorgreys — DISMISSED** [review]

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -554,7 +553,7 @@ def _generate_diagnostic_logs(self):
                 self._log_compat(logging.WARNING, "- DATADOG TRACER DIAGNOSTIC - %s" % msg)
             else:
                 if log.isEnabledFor(logging.INFO):
-                    msg = "- DATADOG TRACER CONFIGURATION - %s" % json.dumps(info)
+                    msg = "- DATADOG TRACER CONFIGURATION - %s" % info
```
Comment (P403n1x87) [inline ddtrace/tracer.py:556]: how confident are we that this won't leak any data that should not be sent to logs?

Comment (therve) [reply ddtrace/tracer.py:556]: No idea.

**Comment by OmerRaviv** [comment]
Related: https://github.com/DataDog/dd-trace-py/issues/4896

**Review by P403n1x87 — APPROVED** [review]

**Review by emmettbutler — APPROVED** [review]

**Pull request merged**